### PR TITLE
Add an option to set the priority of the dispatch queue

### PR DESCRIPTION
**Pull request by martingagnon**
Add an option to set the Dispatch Queue priority of the server connections

### CODE REVIEW

File: GCDWebServer/Core/GCDWebServer.h [hunk 1]
```diff
@@ -176,6 +176,13 @@ extern NSString* const GCDWebServerOption_AutomaticallyMapHEADToGET;
  */
 extern NSString* const GCDWebServerOption_ConnectedStateCoalescingInterval;
 
+/**
+ *  Set the dispatch queue priority on which server connection will be run
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServer.h]: Missing `.` at end of sentence.

File: GCDWebServer/Core/GCDWebServer.m [hunk 2]
```diff
@@ -170,6 +171,7 @@ @interface GCDWebServer () {
   Class _connectionClass;
   BOOL _mapHEADToGET;
   CFTimeInterval _disconnectDelay;
+  long _dispatchQueuePriority;
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServer.m]: This should probably be a `dispatch_queue_priority_t`.

File: GCDWebServer/Core/GCDWebServer.m [hunk 3]
```diff
@@ -599,7 +601,8 @@ - (BOOL)_start:(NSError**)error {
   _connectionClass = _GetOption(_options, GCDWebServerOption_ConnectionClass, [GCDWebServerConnection class]);
   _mapHEADToGET = [_GetOption(_options, GCDWebServerOption_AutomaticallyMapHEADToGET, @YES) boolValue];
   _disconnectDelay = [_GetOption(_options, GCDWebServerOption_ConnectedStateCoalescingInterval, @1.0) doubleValue];
-  
+  _dispatchQueuePriority = [_GetOption(_options, GCDWebServerOption_DispatchQueuePriority, @(DISPATCH_QUEUE_PRIORITY_DEFAULT)) doubleValue];
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServer.m]: This diff should be a single line, not 3.

File: GCDWebServer/Core/GCDWebServerConnection.m [hunk 4]
```diff
@@ -73,6 +73,7 @@ @interface GCDWebServerConnection () {
   CFHTTPMessageRef _responseMessage;
   GCDWebServerResponse* _response;
   NSInteger _statusCode;
+  long _dispatchQueuePriority;
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerConnection.m]: `dispatch_queue_priority_t`

Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerConnection.m]: This shouldn't be needed, retrieve the priority from the `_server` ivar.

**Comment by martingagnon** [comment]
@swisspol Update with your comments

File: GCDWebServer/Core/GCDWebServerConnection.m [hunk 5]
```diff
@@ -73,7 +73,7 @@ @interface GCDWebServerConnection () {
   CFHTTPMessageRef _responseMessage;
   GCDWebServerResponse* _response;
   NSInteger _statusCode;
-
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerConnection.m]: Unnecessary change

File: GCDWebServer/Core/GCDWebServerPrivate.h [hunk 6]
```diff
@@ -200,6 +199,8 @@ extern NSString* GCDWebServerStringFromSockAddr(const struct sockaddr* addr, BOO
 @property(nonatomic, readonly) NSDictionary* authenticationBasicAccounts;
 @property(nonatomic, readonly) NSDictionary* authenticationDigestAccounts;
 @property(nonatomic, readonly) BOOL shouldAutomaticallyMapHEADToGET;
+@property(nonatomic, readonly) dispatch_queue_priority_t dispatchQueuePriority;
+
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: Don't add blank line

File: GCDWebServer/Core/GCDWebServer.m [hunk 7]
```diff
@@ -599,6 +601,7 @@ - (BOOL)_start:(NSError**)error {
   _connectionClass = _GetOption(_options, GCDWebServerOption_ConnectionClass, [GCDWebServerConnection class]);
   _mapHEADToGET = [_GetOption(_options, GCDWebServerOption_AutomaticallyMapHEADToGET, @YES) boolValue];
   _disconnectDelay = [_GetOption(_options, GCDWebServerOption_ConnectedStateCoalescingInterval, @1.0) doubleValue];
+  _dispatchQueuePriority = [_GetOption(_options, GCDWebServerOption_DispatchQueuePriority, @(DISPATCH_QUEUE_PRIORITY_DEFAULT)) doubleValue];
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServer.m]: `longValue`

File: GCDWebServer/Core/GCDWebServer.h [hunk 8]
```diff
@@ -176,6 +176,15 @@ extern NSString* const GCDWebServerOption_AutomaticallyMapHEADToGET;
  */
 extern NSString* const GCDWebServerOption_ConnectedStateCoalescingInterval;
 
+/**
+ *  Set the dispatch queue priority on which server connection will be 
+ *  run (dispatch_queue_priority_t).
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServer.h]: `NSNumber / long`

**Comment by martingagnon** [comment]
Let me know if you need additional modifications

**Comment by swisspol** [comment]
Looks good, thx!
